lib/member_page.rb

```diff
@@ -31,26 +31,31 @@ class MemberPage < Scraped::HTML
   end
 
   field :electoral_history do
-    eh = noko.xpath('//td[contains(.,"Electoral History")]/following-sibling::td')
-    terms = eh.first
-              .text
-              .scan(/([A-Z][a-z]+(?:st|nd|rd|th)\sParliament\s?:?)/)
-              .flatten
-              .map(&:tidy)
-
-    data = eh.text
-             .split(Regexp.union(terms))
-             .map(&:tidy).reject(&:empty?)
-             .map{ |str| str.delete(':') }
-             .map { |str| str.split(/(\d\d?\.\d\d?\.\d\d(?:\d\d)?)/).map(&:tidy) }
-             .map { |arr| Hash[*arr] }
-    terms.map { |str| str.delete(':') }
-         .map(&:tidy)
-         .zip(data).to_h
+    data = electoral_history_section.text
```

Comment:
This is all much too complex to understand here. I suggest you make the electoral history section a `fragment`, so you can have different methods operating on it.